app/routes.py

```diff
@@ -0,0 +1,337 @@
+
+def validate_customer_request_body(request_body):
+    if "name" not in request_body:
+        return jsonify({"details": "Request body must include name."}), 400
+    if "postal_code" not in request_body:
+        return jsonify({"details": "Request body must include postal_code."}), 400
+    if "phone" not in request_body:
+        return jsonify({"details": "Request body must include phone."}), 400
+    return False
+
+def validate_video_request_body(request_body):
+    if "title" not in request_body:
+        return jsonify({"details": "Request body must include title."}), 400
+    if "release_date" not in request_body:
+        return jsonify({"details": "Request body must include release_date."}), 400
+    if "total_inventory" not in request_body:
+        return jsonify({"details": "Request body must include total_inventory."}), 400
+    return False
+
+# ---------------------------
+# ------- CUSTOMERS ---------
+# ---------------------------
+
+# Posts new customer
+@customers_bp.route("", methods = ["POST"])
+def create_customer():
+    request_body = request.get_json()
+    # Check if request_body is invalid/missing data
+    invalid = validate_customer_request_body(request_body)
+    if invalid:
+        # Returns the invalid error message and status code
+        return invalid
+        
+    new_customer = Customer(name=request_body["name"],
+                    postal_code=request_body["postal_code"],
+                    phone=request_body["phone"])
+    
+    db.session.add(new_customer)
+    db.session.commit()
+
+    new_customer_response = new_customer.to_dict()
+    return jsonify(new_customer_response), 201
+
+# Handles all customers
+@customers_bp.route("", methods=["GET"])
+def handle_customers():
+    customers_response = []
+    sort_by = request.args.get('sort')
+    # Poss. Refactor: Could make a helper function, passing in sort_by as parameter
+    if sort_by == "asc":
+        customers = Customer.query.order_by(Customer.name).all()
+    elif sort_by == "desc":
+        customers = Customer.query.order_by(desc(Customer.title)).all()
+    else:
+        customers = Customer.query.all()
+    for customer in customers:
+        customers_response.append(customer.to_dict())
+    return jsonify(customers_response), 200
+
+# Handles one customer
+@customers_bp.route("/<id>", methods=["GET", "PUT"])
+def handle_customer(id):
+    id = validate_id_int(id)
+    customer = Customer.query.get(id)
+    if not customer:
+        return make_response({"message": f"Customer {id} was not found"}, 404)
+    if request.method == "GET":
+        return jsonify(customer.to_dict()), 200
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        # Check if request_body is invalid/missing data
+        invalid = validate_customer_request_body(request_body)
+        if invalid:
+            # Returns the invalid error message and status code
+            return invalid
```

Comment:
I just want to point out that this helper is useful, but the following if statement is a bit awkward.  There are ways to do what you're doing here with a decorator and then you don't need to worry about validation in the body of the function.  

Not something to worry about at Ada , but something to think about for future learning.

Reply:
Thanks for bringing that up! It honestly didn't cross my mind that a decorator could be used for this.